virtool/blast/task.py

```diff
@@ -64,34 +64,56 @@ async def wait(self):
         the search completes. The task completes when either an error or
 
         """
-        try:
-            while True:
-                await asyncio.sleep(self.interval)
 
-                blast = await self.data.blast.check_nuvs_blast(
-                    self.analysis_id, self.sequence_index
-                )
+        for _ in range(0, retries):
+            try:
+                while True:
+                    await asyncio.sleep(self.interval)
 
-                if blast.ready:
-                    logger.info("Retrieved result for BLAST %s", blast.rid)
-                    break
+                    blast_coroutine = self.data.blast.check_nuvs_blast(
+                        self.analysis_id, self.sequence_index
+                    )
+
+                    blast = await asyncio.wait_for(blast_coroutine, 600)
```

Comment:
I like the usage of `wait_for` but in this case I don't think this will work the way you intended. This will only timeout if a single check takes more than 10 minutes. Would be more appropriate if it was wrapping the the whole checking loop. (An individual check should take seconds at most, what we want to do is retry the blast if the remote run isn't done after 10 minutes)